langkit/support/langkit_support-token_data_handlers.adb

```diff
@@ -170,11 +168,37 @@ package body Langkit_Support.Token_Data_Handlers is
       Idx : Natural := 0;
       --  Index in T of the newline character that ends the currently processed
       --  line.
+
+      function Index
+        (Buffer             : Text_Type;
+         Char               : Wide_Wide_Character;
+         Start_Pos, End_Pos : Positive) return Natural;
+      --  Helper function. Return first occurence of ``Char`` in ``Buffer
```

Comment:
Return the index* of the first occurence […]. I think this should also mention that it returns 0 if there is no such occurence.

BTW, what about instead changing the declaration of `T` above so that it covers only the part of `Source_Buffer` that is initialized, and remove the `End_Pos` argument here?

Reply:
> BTW, what about instead changing the declaration of T above so that it covers only the part of Source_Buffer that is initialized, and remove the End_Pos argument here?


I don't see the point. I must pass Start_Pos anyway.